.github/workflows/integration_tests.yml

```diff
@@ -0,0 +1,46 @@
+name: Nightly integration tests
+
+# Run on request and every day at 3 AM UTC
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: 0 3 * * *
+
+jobs:
+  quantinuum:
+    name: Quantinuum (simulator)
+    runs-on: ubuntu-latest
+    container:
+      image: ghcr.io/${{ github.repository }}:latest
+      options: --user root
+
+    steps:
+      - name: Get code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 2
+
+      - name: Set git safe
+        run: git config --global --add safe.directory $PWD
+
+      - name: Install requirements
+        run: |
+          apt-get update
+          apt-get install -y curl jq
+
+      - name: Setup quantinum account
+        run: |
+          curl -X POST -H "Content Type: application/json" -d '{ "email":"${{ secrets.BACKEND_LOGIN_EMAIL }}","password":"${{ secrets.QUANTINUUM_PASSWORD }}" }' https://qapi.quantinuum.com/v1/login > credentials.json
+          id_token=`cat credentials.json | jq -r '."id-token"'`
+          refresh_token=`cat credentials.json | jq -r '."refresh-token"'`
+          echo "key: $id_token" > ~/.quantinuum_config
+          echo "refresh: $refresh_token" >> ~/quantinuum_config
+
+      - name: Build and run
+        run: |
+          for filename in test/NVQPP/integration/*.cpp; do
+            [ -e "$filename" ] || continue
```

Comment:
Not redundant. In bash, when one tries to iterate over files by globbing (which is the case), it's good practice to avoid the corner case where the glob does not match. The problem is that when the glob doesn't match, loop variable expand to the (un-matching) glob pattern string itself, and thus we would try to compile `test/NVQPP/integration/*.cpp`

Reply:
Sidebar random note:

An alternative to using globbing could be something like
```sh
for filename in `find test/NVQPP/integration -maxdepth 1 -name '*.cpp'`; do
```
(Not tried under the .yml context here, so YMMV.)

Not sure which, if either, is more appealing.

Reply:
I would prefer to fail if we don't find any tests here. The expectation is that we run some tests, and if someone moves the folder without updating the workflow, then this workflow should fail.

Reply:
> I would prefer to fail if we don't find any tests here. The expectation is that we run some tests, and if someone moves the folder without updating the workflow, then this workflow should fail.

Agreed. I made changes so that instead of continuing, the loop will emit an error if the glob fails. This way if we move these files, we won't have the problem of always passing this workflow without actually testing anything.